cpp/src/parquet/arrow/reader.h

```diff
@@ -21,6 +21,8 @@
 #include <memory>
 #include <vector>
 
+#include "arrow/util/async_generator.h"
```

Comment:
This is a heavy include, can we avoid adding it?

Reply:
I think we've run into this before - with the current definition of AsyncGenerator we can't easily provide a forward declaration, right?

Reply:
Hmm, then we can simply use the actual expansion, i.e. `std::function<Future<T>()>`.